lib/api-util.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+
+var preq = require('preq');
+var sUtil = require('../lib/util');
+var Template = require('swagger-router').Template;
+
+var HTTPError = sUtil.HTTPError;
+
+
+/**
+ * Calls the MW API with the supplied query as its body
+ *
+ * @param {Object} app the application object
+ * @param {string} domain the domain to issue the request to
+ * @param {Object} query an object with all the query parameters for the MW API
+ * @return {Promise} a promise resolving as the response object from the MW API
+ */
+function mwApiGet(app, domain, query) {
+
+    query = query || {};
+    query.continue = query.continue || '';
+
+    var request = new Template(app.conf.mwapi_req).expand({
```

Comment:
Hm.. Why are we recompiling the template on each request? It's incredibly expensive and slow